src/XMakeCommandLine/XMake.cs

```diff
@@ -2835,10 +2835,7 @@ string unquotedParameter
             {
                 logger = loggerDescription.CreateLogger();
 
-                if (logger == null)
-                {
-                    InitializationException.VerifyThrow(logger != null, "LoggerNotFoundError", unquotedParameter);
-                }
+                InitializationException.VerifyThrow(logger != null, "LoggerNotFoundError", unquotedParameter);
```

Comment:
But regardless of how `CreateLogger` might change, the null check is duplicated in the `if` and the `VerifyThrow` argument, right?